### PR TITLE
Add masOS compatibility for 'sed' usage in scripts

### PR DESCRIPTION
**Pull request by PhilipSchmid**

### CODE REVIEW

File: api/export-doc.sh
Comment (mtardy) [inline api/export-doc.sh:29]: So I did not experience the backup file that might be created, if that happens, are you sure about its name (the `-e` suffix)?

Anyway could you suppress the error from rm `rm: cannot remove '/tmp/tmp.V2HaNn1uox-e': No such file or directory` because it looks something wrong happened

```suggestion
rm ${TMP_FILE}-e 2>/dev/null || true
```

File: install/kubernetes/export-doc.sh
Comment (mtardy) [inline install/kubernetes/export-doc.sh:31]: same here

File: api/export-doc.sh
Comment (PhilipSchmid) [reply api/export-doc.sh]: > So I did not experience the backup file that might be created

On macOS or Linux? I've tested on macOS and did not get this `rm: cannot remove` error. Whatever, I've added the `2>/dev/null` as I think it makes sense either way. Thanks!

**Comment by mtardy** [comment]
empty line issue, just run the script locally and push!

**Review by mtardy — CHANGES_REQUESTED** [review]
previous
```
# cleanup unecessary consecutive whitelines
sed -i '/^[[:space:]]*$/N;/^[[:space:]]*\n[[:space:]]*$/D' $TMP_FILE
```
was working on macOS and Linux for me.

It doesn't work for you? can you dump your sed version maybe?

**Comment by PhilipSchmid** [comment]
```bash
# cleanup unecessary consecutive whitelines
sed -i'' -e '/^[[:space:]]*$/N;/^[[:space:]]*\n[[:space:]]*$/D' $TMP_FILE
```

It works and does what it should do (`cleanup unecessary consecutive whitelines`) but there's no empty line at the end of the `docs/content/en/docs/reference/helm-chart.md` file on macOS, while there is one Linux. Hence, I'm still searching for a solution that produces the same output for both.

**Review by mtardy — CHANGES_REQUESTED** [review]
this doesn't run the same in the CI and on your platform since `git status` returns something. You can't reproduce?

**Comment by PhilipSchmid** [comment]
Rebased to `main`.

@mtardy Exactly, I'm not able to reproduce this locally. `cd install/kubernetes && ./test.sh` doesn't produce any changes on my macOS or my Linux server 🤷‍♂️.

**Comment by jrfastab** [comment]
going to give this a small nudge. Would be nice if MacOS works also.

**Review by mtardy — APPROVED** [review]
Thanks Philip, sorry for the length but those kinds of compatibility things are a bit annoying as you saw 😛!

I think the generated file test passes now, don't know what changed but it looks good! Let's merge this when the whole CI passes.

File: install/kubernetes/export-doc.sh
Comment (mtardy) [inline install/kubernetes/export-doc.sh:50]: Oh lookout for this before merging, this got upgraded and this is reversing the change

Comment (PhilipSchmid) [reply install/kubernetes/export-doc.sh]: Good catch! Updated, thanks!

**Comment by mtardy** [comment]
Thanks!! 🎉

**Pull request merged**
